src/sentry/workflow_engine/endpoints/validators/detector_workflow.py

```diff
@@ -9,6 +11,31 @@
 from sentry.workflow_engine.models.workflow import Workflow
 
 
+def check_can_edit_detector(detector: Detector, request: Request) -> None:
```

Comment:
I think this would read better if it returned a boolean and the places that call it can raise the permissions error if it's false